spidermon/contrib/scrapy/monitors.py

```diff
@@ -522,6 +530,78 @@ def test_execution_time(self):
         self.assertLess(duration.total_seconds(), max_execution_time, msg=msg)
 
 
+@monitors.name("Jobs Comparison Monitor")
+class JobsComparisonMonitor(BaseStatMonitor):
+    """
+    .. note::
+       This monitor is useful when running jobs in
+       `Zyte's Scrapy Cloud <https://www.zyte.com/scrapy-cloud/>`_.
+
+    Check for a drop in scraped item count compared to previous jobs.
+
+    You need to set the number of previous jobs to compare, using ``SPIDERMON_JOBS_COMPARISON``.
+    The default is ``0`` which disables the monitor. We use the average of the scraped items count.
+
+    You can configure which percentage of the previous item count is the minimum acceptable, by
+    using the setting ``SPIDERMON_JOBS_COMPARISON_THRESHOLD``. We expect a float number between
+    ``0.0`` and ``1.0``. The default is ``0.8``.
+
+    You can filter which jobs to compare based on their states using the
+    ``SPIDERMON_JOBS_COMPARISON_STATES`` setting. The default value is ``("finished",)``.
+
+    You can also filter which jobs to compare based on their tags using the
+    ``SPIDERMON_JOBS_COMPARISON_TAGS`` setting. Among the defined tags we consider only those
+    that are also present in the current job.
+    """
+
+    stat_name = "item_scraped_count"
+    assert_type = ">="
+
+    def _get_jobs(self, states, number_of_jobs, tags=None):
+        return zyte.client.spider.jobs.list(
+            state=states,
+            count=number_of_jobs,
```

Comment:
I think this would only return a maximum of `1000` jobs because `jobs.list` uses `jobs.iter` internally and it only returns 1000 results by default
> [by default :meth:`Jobs.iter` returns maximum last 1000 results](https://github.com/scrapinghub/python-scrapinghub/blob/master/scrapinghub/client/jobs.py#L189)

cc: @rennerocha